src/util/logging.cpp

```diff
@@ -41,6 +41,9 @@ enum class WriteFlag {
 
 Q_DECLARE_FLAGS(WriteFlags, WriteFlag)
 
+#if defined(__clang__)
+#pragma clang diagnostic ignored "-Wunused-function"
```

Comment:
This applies to the whole file. Can we just do it for the single function?

```cpp
#pragma clang diagnostic push
#pragma clang diagnostic ignored "-Wunused-function"
// Code that throws warning goes here
#pragma clang diagnostic pop
```

Also, please add a comment why this is necessary.

Reply:
Another thing: It this warning legit? If so, I assume that GCC doesn't throw the warning because it's buggy. If that's the case, it might get fixed in later versions of GCC. Hence, we could use this instead:

```
#pragma GCC diagnostic push
#pragma GCC diagnostic ignored "-Wunused-function"
// Code that throws warning goes here
#pragma GCC diagnostic pop
```

Apparently clang understands GCC pragmas, but not vice versa.

Reply:
I don't want to complicate the code unless actually needed. This could happen whenever using Q_DECLARE_OPERATORS_FOR_FLAGS, which defines functions that might never be used.